### ISSUE TITLE
Higher vf call AD values after using vg filter

### ISSUE THREAD

**Issue by danielwood1992** [vgteam/vg#4443]
**1. What were you trying to do?**

```
vg giraffe --output-format gam --gbz-name $vg.gbz --minimiser-name $autoindex.min --dist-name $autoindex.dist -f $R1 -f $R2 -t $threads $reference.fa $vcf.gz > $gam
vg filter $gam -r 0.9 -fu -m 1 -q 20 -D 999 -x $vg.gbz > $filter.gam
vg pack -t $threads -x $vg.xg -g $filter.gam -Q 5 -o $filter.gam.pack
vg call $vg.xg -t $threads -k $filter.gam.pack -v $vcf > $filter.gam.pack.vcf

vg pack -t $threads -x $vg.xg -g $gam -Q 5 -o $gam.pack
vg call $vg.xg -t $threads -k $gam.pack -v $vcf > $gam.pack.vcf
```

**2. What did you want to happen?**

Identify coverage of ref/alt alleles from the vcf call AD statistic - and looking at the impact of the vg filter/call step.

**3. What actually happened?**

Running vg filter on the gam reduced the size of the gam substantially. In most cases the AD for each site in $gam.pack.vcf vs. $filter.gam.pack.vcf made sense: the AD was higher for the calls from the unfiltered gam. However there a minority of sites (~2.5%) of cases where the AD for either the reference or alt allele is higher in the filtered gam than the same site in the unfiltered gam (which substantially alters the estimated ref/alt allele coverage - see example graph 
![AF_Graph](https://github.com/user-attachments/assets/5abf3edc-de29-4def-a918-b7ab7aa5ae2f)
, blue points are where the ref or alt allele coverage is higher from the filtered vcf, red are where it is lower) - I don't understand how this can be the case, surely the filtering should only be removing reads? The graph is made from a biallelic vcf so it shouldn't be mutli-allele mapping that is the problem. 

**5. What data and command can the vg dev team use to make the problem happen?**
I can send the data and exact script I used if needed. Any thoughts on this much appreciated!

**6. What does running `vg version` say?**

vg version v1.48.0 "Gallipoli"
Compiled with g++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0 on Linux
Linked against libstd++ 20210601```

```

**Comment by glennhickey** [comment]
Yeah, that doesn't seem to make much sense.  Are you able to share the gam and gbz, along with a site that's identical in the vcf but has higher AD in the filtered one?  Thanks.

**Comment by danielwood1992** [comment]
Yes I can - do you have an email address I can send a google drive link to? Thanks a lot!

**Comment by glennhickey** [comment]
Sure, use my name here, glennhickey, at gmail.

**Comment by danielwood1992** [comment]
great - shared, let me know if you have trouble accessing

**Comment by glennhickey** [comment]
got it, thanks.

**Comment by glennhickey** [comment]
Actually, I don't see the .xg in the shared folder, which is required for `call -v`.

**Comment by dwood92** [comment]
Ah yes sorry - that should be in there now

**Comment by glennhickey** [comment]
Hi, sorry for the delay.  I have indeed reproduced this and agree there is something fishing going on.  If the genotype is different, then it's understandable for AD to be different due to how call uses rounding and flow assignment to come up with support.  But there are cases with the same genotype with completely different weights that I can't explain.  Anyway, I haven't forgotten, just haven't had time to get to the bottom of it -- will hopefully have news soon.

**Comment by danielwood1992** [comment]
Hi Glen, thanks for the update: glad it's not just me! Let me know if there's any other files you need here.

**Comment by danielwood1992** [comment]
Hi Glen: did you figure out what was causing this issue in the end? If you've downloaded the files I might delete them from my google drive as they're taking up a lot of space!

**Comment by glennhickey** [comment]
Goodness, sorry for not getting back to you.  I was looking at a similar issue and thought it was related but it wasn't.  And what's worse, I seem to have lost your data.  

While the drive link you sent still seems up, the GAM doesn't seem to be in it any more?

**Comment by danielwood1992** [comment]
Hi Glenn,

No problem - thanks a lot for getting back to me. I ended up deleting the gam files from that folder - I will re-upload those, I will let you know when they’re available again. 

I hope you don’t mind if I add some wider context: I am primarily trying to use vg call to get allele frequencies of SVs from poolseq data, as well as demonstrate we can use it to call SVs from individually sequenced samples as well. So the most critical thing here is for me to be able to get the number of reads supporting each individual. I’ve not been able to figure out a way of just getting the reads supporting the alternate vs. reference allele in the vcf, regardless of a genotype call - would there be a better way of doing this than vg call, do you think? I had tried using the options -B -m 1,1 -b 50,50 to try and make it call everything as a heterozygote, but this doesn’t seem to have worked. 

In any case, when I generate an in silico pool from a set of 42 individuals, and compare the genotype estimates from the individuals (X axis in graph attached) vs. the allelic depth from doing vg call on the pool (Y axis), I get a lot of sites that are estimated as low frequency in the individually genotyped samples, but then a very high frequency in the in silico pool. For many of these sites, I’m not sure the genotypes for individually sequenced samples make much sense either - I get plenty of cases where, for example, where the genotype:depth:allelic depth looks like this -  0/0:16:3,13. I don’t know whether it’s sometimes getting confused as to which is the alternate path and which is the reference? When I asked my original question I was changing the filtering parameters to see if that made a difference: but as you’ve seen, there are cases where the AD numbers just don’t seem to make sense here either.

![Image](https://github.com/user-attachments/assets/5a72b8e2-92db-4620-a105-97afe88f7a83)

The other thing to note is that I was previously using vg giraffe to map the reads, which I thought might be affected by the fact my vcf isn’t phased- I only have biallelic SVs, so I thought it would be ok to create an index with vg gbwt –path-cover as there shouldn’t be tons of alternate haplotypes over the distance a short read would map. In any case, it doesn’t seem like the strange genotypes I get are restricted to SVs near other SVs. Plus I have been redoing the mapping using vg map, which does seem to improve my average mapping quality - but these sites still seem to have a lot of weird genotype calls, so I’m not sure it’s the main culprit, but thought I should bring it up anyway. 

Any thoughts you have on any of this would be much appreciated - and let me know if you’d like any of the files from the other problems I’ve described. 

Thanks a lot and best wishes,

Daniel Wood

**Comment by danielwood1992** [comment]
Hi Glenn, those gam files are back in the directory now. I have also put the vcfs in there for both as well. An example of a site with that sort of allele frequency in the unfiltered gam file is Scf9YQZ_9_HRSCAF_17     31983451        0/0:18:4,14. Perhaps an earlier version of vg might be more suitable for me then - perhaps 1.25?

**Comment by glennhickey** [comment]
Got the GAMs, but don't see the xg

**Comment by danielwood1992** [comment]
should be in there now!

**Comment by glennhickey** [comment]
@danielwood1992 Would you mine please trying this patched vg to see if it helps your issue?

https://public.gi.ucsc.edu/~hickey/vg-patch/vg.2c61112bb461e3a8e6b976baefe464798ef3f359

**Comment by danielwood1992** [comment]
Hi Glenn, that's great, thanks very much - I am rerunning my pipeline now so I will let you know how it goes.

**Comment by danielwood1992** [comment]
Hi Glenn - the patch seems to have fixed the problem of homozygous reference calls with very weird allele depths, so that's great. However, I still have the problem of SV calls from pooled sequencing data having a substantial number of sites with much higher allele depths in a pool than the allele frequencies for individuals that went into the pool would suggest (see image - the top left corner is the problem) - and unfortnately when doing a GWAS with multiple pools, the sites that are significant are highly overrepresented in these strangely behaving SVs. There's nothing as obviously wrong with these calls as with those with a clearly incorrect genotype. I'm not sure whether this is an issue with getting the alleles supporting the ref/alt alleles in the pool by making it call a genotype -  do you know if there's a way I can just get the relative depth support for the alt/ref alleles in the vcf, without calling a genotype? Any thoughts you have would be much appreciated!

<img width="445" alt="Image" src="https://github.com/user-attachments/assets/6c10ace2-a0e3-428c-8b6d-21a421183176" />

**Comment by glennhickey** [comment]
Don't have any great ideas.... Is your VCF bi-allelic (one alt / site)?  I could look into hacking an option where it just sets every GT to `0/1`, with the supports output accordingly.  Do you think that would help?
